### PR TITLE
Add automated tests for custom vs listeners

### LINKED ISSUE #4196: Add new python tests for custom listener controls

### PR DESCRIPTION
**Pull request by haywoodsh**
### Proposed changes

Add automated test for custom listeners for virtual servers #4271 
closes #4196 

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by vepatel — APPROVED** [review]

File: tests/suite/test_virtual_server_custom_listeners.py [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+            assert "listen 8445 ssl;" in vs_config
+            assert "listen [::]:8445 ssl;" in vs_config
+        else:
+            assert "listen 8445 ssl;" not in vs_config
+            assert "listen [::]:8445 ssl;" not in vs_config
+
+        assert "listen 80;" not in vs_config
+        assert "listen [::]:80;" not in vs_config
+        assert "listen 443;" not in vs_config
+        assert "listen [::]:443;" not in vs_config
```
Comment (shaun-nx) [inline tests/suite/test_virtual_server_custom_listeners.py:206]:
```suggestion
        assert "listen 443 ssl;" not in vs_config
        assert "listen [::]:443 ssl;" not in vs_config
```

File: tests/suite/test_virtual_server_custom_listeners.py [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
```
Comment (shaun-nx) [inline tests/suite/test_virtual_server_custom_listeners.py:351]:
```suggestion
        assert "listen 443 ssl;" not in vs_config
        assert "listen [::]:443 sll;" not in vs_config
```

**Review by shaun-nx — CHANGES_REQUESTED** [review]

**Review by shaun-nx — APPROVED** [review]
👍

File: tests/suite/test_virtual_server_custom_listeners.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+
+        print("\nStep 4: Test HTTP responses")
+        urls = [
+            virtual_server_setup.backend_1_url,
+            virtual_server_setup.backend_1_url_ssl,
+            virtual_server_setup.backend_1_url_custom,
+            virtual_server_setup.backend_1_url_custom_ssl,
+        ]
+        for url, expected_response in zip(urls, test_setup["expected_response_codes"]):
+            if expected_response > 0:
```
Comment (vepatel) [inline tests/suite/test_virtual_server_custom_listeners.py:216]: iterable type(string) to int needed

File: tests/suite/test_virtual_server_custom_listeners.py [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+        wait_before_test()
+
+        urls = [
+            virtual_server_setup.backend_1_url,
+            virtual_server_setup.backend_1_url_ssl,
+            virtual_server_setup.backend_1_url_custom,
+            virtual_server_setup.backend_1_url_custom_ssl,
+        ]
+
+        for expected_response, url in zip(urls, [404, 404, 200, 200]):
```
Comment (vepatel) [inline tests/suite/test_virtual_server_custom_listeners.py:303]:
```suggestion
        for url, expected_response in zip(urls, [404, 404, 200, 200]):
```
@haywoodsh

**Review by vepatel — APPROVED** [review]

**Pull request merged**